### PR TITLE
Validation and spec

### PR DESCRIPTION
**Pull request by fatima-najafi**
### **Project requirements**
Add the following validations:
For the User model:
Name must not be blank.
PostsCounter must be an integer greater than or equal to zero.
For the Post model:
Title must not be blank.
Title must not exceed 250 characters.
CommentsCounter must be an integer greater than or equal to zero.
LikesCounter must be an integer greater than or equal to zero.
Add unit specs for all of models' methods and validations.

### CODE REVIEW

File: spec/post_spec.rb [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+  it 'should be valid when likes_counter is positive' do
+    subject.likes_counter = 5
+    expect(subject).to be_valid
+  end
+
+  # test recent_comment
+  it 'should have a recent_comments method' do
+    expect(first_post).to respond_to(:recent_comments)
+  end
```
Comment (danmainah) [inline spec/post_spec.rb]: - [x] @fatima-najafi Great work adding a test to validate the presence of the recent_comments method, however we are suppose to test if its working as expected if when you have more than 5 comments for the post it only display the 5 recent comments.

File: spec/user_spec.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  end
+  # Test most recent posts............
+
+  it 'returns the most recent 3 posts' do
+    subject.posts.create(title: 'Post 1', text: 'This is the first post.', comments_counter: 0, likes_counter: 0)
+    subject.posts.create(title: 'Post 2', text: 'This is the second post.', comments_counter: 0, likes_counter: 0)
+    subject.posts.create(title: 'Post 3', text: 'This is the third post.', comments_counter: 0, likes_counter: 0)
+    recent_posts = subject.recent_posts
+    expect(recent_posts.length).to eq(3)
+  end
```
Comment (danmainah) [inline spec/user_spec.rb]: - [x] @fatima-najafi Great work adding a test to check if the recent_posts method is working correctly, however giving it onlu3 posts won't effectively check if it's working correctly, kindly consider adding like 5 posts and check if it only displays 3 recent posts.

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @fatima-najafi , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️  Added validation for the user and post models
✔️  Added tests for the models' validation and custom methods


# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by zieeco — APPROVED** [review]
# status: APPROVED :green_circle: :100: 

Hi, @fatima-najafi, 


68747470733a2f2f6d656469612e67697068792e636f6d2f6d656469612f78543078657a514755357843444a754350652f67697068792e676966


Good job so far 👍 Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights ⚡ 

✅ Descriptive PR 👍 
✅ Requested changes implemented 🔥 


## Optional suggestions
- N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._



Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**